tests/unittest/_torch/sampler/test_logits_logprobs.py

```diff
@@ -0,0 +1,605 @@
+            if sampling_params_list[req_idx]._greedy_decoding:
+                probs = torch.zeros_like(logits_for_token)
+                probs[0, token_ids[token_idx]] = 1.0
+            else:
+                topk = topk if topk is not None else logits_for_token.shape[-1]
+                topp = topp if topp is not None else 1.0
+                temperature = temperature if temperature is not None else 1.0
+
+                # perform maksing top-k top-p
+                if simple_llm.args.disable_flashinfer_sampling:
+                    _, probs = top_k_top_p_sampling_batch(
+                        logits_for_token,
+                        top_k=topk,
+                        top_p=topp,
+                        temperature=temperature)
+                else:
+                    logits_for_token = logits_for_token.to(device="cuda")
+                    _, probs = _StrategyImpls.StrategyImplWithProbs._sample_with_probs(
+                        logits_for_token,
+                        group_logit_indices=None,
+                        top_k=torch.tensor([topk],
+                                           dtype=torch.int32,
+                                           device="cuda"),
+                        top_p=torch.tensor([topp],
+                                           dtype=torch.float32,
+                                           device="cuda"),
+                        temperature=torch.tensor([temperature],
+                                                 dtype=torch.float32,
+                                                 device="cuda"),
+                        generator=None)
+
+            if temperature != 0:
+                logits_for_token /= temperature
+            adjusted_logits_for_token = torch.where(probs != 0,
+                                                    logits_for_token,
+                                                    float("-inf"))[0]
+            expected_logprobs = torch.nn.functional.log_softmax(
+                adjusted_logits_for_token, dim=-1).to(device="cpu")
+            for logprob_token, logprob_values in token_logprobs_dict.items():
+                expected_logprob = expected_logprobs[logprob_token].item()
+                returned_logprob = logprob_values.logprob
+                logprob_diff = abs(returned_logprob - expected_logprob)
+                if returned_logprob == float(
+                        "-inf") and expected_logprob == float("-inf"):
+                    logprob_diff = 0
+                print(
+                    f"Req {req_idx}, Token {token_idx} ({logprob_token}): returned={returned_logprob:.6f}, expected={expected_logprob:.6f}, diff={logprob_diff:.6f}"
+                )
+
+                assert logprob_diff < 1e-4, \
+                    f"Req {req_idx}, Token {token_idx}: Logprob mismatch! " \
+                    f"Returned {returned_logprob:.6f} but expected {expected_logprob:.6f} " \
+                    f"(diff={logprob_diff:.6f})."
```

Comment:
Consider [`torch.testing.assert_close`](https://docs.pytorch.org/docs/stable/testing.html#torch.testing.assert_close) instead of manual comparing.